cmd/bootstrap/gr/recovery_method.go

```diff
@@ -114,25 +123,25 @@ func compareGTIDs(ctx context.Context, shell SQLRunner, a, b string) (GTIDSetRel
 
 // If purged has more gtids than the executed on the replica
 // it means some data will not be recoverable
-func comparePrimaryPurged(ctx context.Context, shell SQLRunner, purged, executed string) bool {
-	query := fmt.Sprintf("SELECT GTID_SUBTRACT('%s', '%s') = ''", purged, executed)
+func comparePrimaryPurged(ctx context.Context, shell SQLRunner, purged, executed string) (bool, error) {
+	query := fmt.Sprintf("SELECT GTID_SUBTRACT('%s', '%s') = '' AS is_subset", purged, executed)
 
 	result, err := shell.runSQL(ctx, query)
 	if err != nil {
-		return false
+		return false, errors.Wrap(err, "run sql")
 	}
 
-	v, ok := result.Rows[0]["GTID_SUBTRACT"]
+	v, ok := result.Rows[0]["is_subset"]
 	if !ok {
-		return false
+		return false, errors.Errorf("unexpected output: %+v", result)
 	}
 
-	sub, err := strconv.Atoi(v)
-	if err != nil {
-		return false
+	s, ok := v.(float64)
```

Comment:
why float and not int? as far as i remember `SELECT GTID_SUBTRACT('%s', '%s') = ''` returns 0 or 1